File: fastapi/provider_request.py
"""Provider request construction kept in parity with the Node template."""

from __future__ import annotations


def build_provider_request(
    *,
    input_items: list[object],
    instructions: str,
    model: str,
    previous_response_id: str | None,
) -> dict[str, object]:
    if not isinstance(input_items, list):
        raise TypeError("provider input must be an array")
    if not isinstance(instructions, str) or not instructions.strip():
        raise TypeError("provider instructions must be non-empty text")
    if not isinstance(model, str) or not model.strip():
        raise TypeError("provider model must be non-empty text")
    payload: dict[str, object] = {
        "model": model,
        "instructions": instructions,
        "input": input_items,
        "stream": True,
    }
    if previous_response_id:
        payload["previous_response_id"] = previous_response_id
    return payload

File: node/provider-request.mjs
export function buildProviderRequest({
  input,
  instructions,
  model,
  previousResponseId,
}) {
  if (!Array.isArray(input)) throw new TypeError("provider input must be an array");
  if (typeof instructions !== "string" || !instructions.trim()) {
    throw new TypeError("provider instructions must be non-empty text");
  }
  if (typeof model !== "string" || !model.trim()) {
    throw new TypeError("provider model must be non-empty text");
  }
  return {
    model,
    instructions,
    input,
    ...(previousResponseId ? { previous_response_id: previousResponseId } : {}),
    stream: true,
  };
}
